### ISSUE TITLE
[Re-genesis] Prepare a migration plan for the on-chain Morse state

### ISSUE THREAD

**Issue by Olshansk** [pokt-network/poktroll#144]
## Objective

Prepare a migration strategy from Morse to Shannon related to on-chain state.

## Origin Document

- As an example, [here is](https://github.com/pokt-network/pocket-network-genesis/blob/master/mainnet/genesis.json) the Morse mainnet genesis file
- During re-genesis, we will need a `genesis.json` file that reflects all the on-chain in Morse at some specified block height

## Goals

- Finding a stakeholder that can extra and provide all the on-chain data we need (e.g. POKTscan)
- Beginning to build out the utilities needed to automatically generate this file in the future

## Deliverables

- [x] Identify an owner (via a proposal or other) who can build a tool to extract on-chain data
- [x] Deliver a prototype (e.g. a script) that provides `genesis.json` file for Shannon reflecting the on-chain for Morse

## Non-goals / Non-deliverables

- Figuring out a plan for the `wPOKT` migration
- Selecting the exact block height at which we migrate the data
- Providing the exact `.json` file that'll be used for the migration
- Deciding which key schemes will be used for Shannon (use placeholders)
- Deciding which address schemes/mappings will be used for Shannon (use the best approximation)

## General deliverables

- [ ] **Comments**: Add/update TODOs and comments alongside the source code so it is easier to follow.
- [ ] **Testing**: Add new tests (unit and/or E2E) to the test suite.
- [ ] **Makefile**: Add new targets to the Makefile to make the new functionality easier to use.
- [ ] **Documentation**: Update architectural or development READMEs; use [mermaid](https://mermaid-js.github.io/mermaid/) diagrams where appropriate.

---

**Creator**: @Olshansk 
**Co-Owners**: @RossiNYC

**Comment by h5law** [comment]
I have created a discussion [1] related to the different types of keys we could use in Shannon.

[1] https://github.com/pokt-network/poktroll/discussions/215

**Comment by Olshansk** [comment]
Will work on publishing this publically: https://www.notion.so/buildwithgrove/Morse-Shannon-Migration-89d45a8acbaf4f4dbed607f6a8d40609?pvs=4

**Comment by Olshansk** [comment]
The message below was sent to the team at Grove:

---

_*tl;dr Nothing actionable from anyone right now but starting this channel as we start thinking about Morse -> Shannon migration.*_

Core-Team I've been working on a series of docs to kickoff the migration effort.

There will be A LOT more details to figure out.

This is just a starting point to capture some thoughts/ideas and start aligning across multiple efforts:
- Protocol
- Legal
- Ecosystem
- Stakeholder

I was also aiming to get some visuals in place so we can build intuition of what's involved. Attaching some of the images from the notion docs here.

Migration notion doc: https://www.notion.so/buildwithgrove/Morse-to-Shannon-Migration-173a36edfff6809cb1cbe10827c040de?pvs=4

[Optional] Background on addresses key schemes: https://www.notion.so/buildwithgrove/Background-Addresses-Key-Schemes-175a36edfff680ff8f23fc7599fd9107?pvs=4

Technical `onchain` migration: https://www.notion.so/buildwithgrove/Technical-Migration-from-Morse-Shannon-173a36edfff6800689fad43fd15287cd?pvs=4

Operation `offchain` migration: https://www.notion.so/buildwithgrove/Operational-Migration-from-Morse-Shannon-173a36edfff680cf96f5cbbdb20ef366?pvs=4

Legal `tokenomic` migration: https://www.notion.so/buildwithgrove/Legal-Migration-from-Morse-Shannon-173a36edfff680d180c5f892b3bb71c3?pvs=4

<img width="1599" alt="Screenshot 2025-01-08 at 2 47 12 PM" src="https://github.com/user-attachments/assets/fbcfaef8-582e-40a5-992a-c13bfe359911" />
<img width="1463" alt="Screenshot 2025-01-08 at 2 46 47 PM" src="https://github.com/user-attachments/assets/d1a75549-2382-4bd7-ae9c-2876b70a33fc" />
<img width="1320" alt="Screenshot 2025-01-08 at 2 46 32 PM" src="https://github.com/user-attachments/assets/746f1e01-1ab9-4672-83c0-1370d3fe73d7" />
<img width="592" alt="Screenshot 2025-01-08 at 2 46 20 PM" src="https://github.com/user-attachments/assets/1fe4a8b5-af2b-4ea5-92ed-f6f4a5f3b02a" />

**Comment by Olshansk** [comment]
Conversations are ongoing but the mega notion doc to capture all the others is available here: https://www.notion.so/buildwithgrove/Morse-to-Shannon-Migration-173a36edfff6809cb1cbe10827c040de?pvs=4
